### ISSUE TITLE
Publish as android library artifact with proguard rules

### ISSUE THREAD

**Issue by sockeqwe** [Tickaroo/tikxml#17]
We should bundle the `core` and  `annotation` module into a android library `.aar` and add the proguard rules.

It would make sense to use gradle instead of maven as build system for the whole project ...

**Comment by iNoles** [comment]
Is anyone is working for this?

**Comment by FabianTerhorst** [comment]
I don´t think so. Im sure he would accept a pr.

**Comment by sockeqwe** [comment]
No i'm not working on that, but will defenitely work on that. The question is if I do want to do that in maven or gradle.

The original plan was to stick with maven, and simply push a `.aar` file from maven  (android maven plugin) containing proguard rules and dependencies

**Comment by FabianTerhorst** [comment]
Gradle would require a new publish script, but would also be better compatible with android studio.

**Comment by sockeqwe** [comment]
publish script is not an issue. The main issue is that gradle + java module + annotation processing (i.e. for (annotationprocessingtesting)[https://github.com/Tickaroo/tikxml/tree/master/annotationprocessingtesting]) doesn't work well out of the box.

**Comment by FabianTerhorst** [comment]
thats true, the annotation processor dependencies has to be java modules.

**Comment by FabianTerhorst** [comment]
Maybe the new jack compiler with annotation processing support will change something there?

**Comment by sockeqwe** [comment]
i would prefer to stick with java modules. this parser is not android exclusive. 

All modules are plain java modules (except this android module with proguard rules).

The main problem is that for testing I have a `annotationprocessingtesting` module which depends on the `processor` module (kind of integration test) and here gradle fails (or I didn't have enough time to check how to configure gradle properly with annotation processing).

**Comment by FabianTerhorst** [comment]
you have tested if the new jack compiler can run the tests?

**Comment by FabianTerhorst** [comment]
the jack compiler is supporting ```annotationProcessor 'com.google.dagger:dagger-compiler:2.0' ```for example.

**Comment by FabianTerhorst** [comment]
https://android.googlesource.com/toolchain/jack/+/ub-jack-lang-dev/jack-tests/tests/com/android/jack/error/AnnotationProcessorErrorTest.java

**Comment by sockeqwe** [comment]
i haven't tested jack ...I prefer to use maven or gradle for this project
...

Fabian Terhorst <notifications@github.com> schrieb am Sa., 11. Juni 2016,
20:10:

**Comment by sockeqwe** [comment]
For now I would prefer this: 
https://simpligility.github.io/android-maven-plugin/aar.html

it should be as simple as putting a proguard file into a new maven module that uses this maven plugin

**Comment by FabianTerhorst** [comment]
how you get the project currently well imported into android studio?

**Comment by sockeqwe** [comment]
I use Intellij for plain java projects (like this one) and not Android Studio, then Import project --> root `pom.xml` file

**Comment by FabianTerhorst** [comment]
Okay, Android Studio doesn't support importing a Maven project.
